### ISSUE TITLE
Webrender consistently borks fonts

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#11355]
![screenshot from 2016-05-24 10 46 18](https://cloud.githubusercontent.com/assets/1617736/15492873/74d3049e-219d-11e6-84dc-854394c6308f.png)

Note the missing characters, squished text, etc.

This happens on basically every website, but the degree of borking varies.

Possibly related: https://github.com/servo/servo/issues/10020 , https://github.com/servo/servo/issues/10233 , https://github.com/servo/servo/issues/10228

I've not heard of this happening on mac, but it happens on my Ubuntu linux (as well as eddy's NixOS linux) and @metajack's Windows.

**Comment by Manishearth** [comment]
```
$ lspci | grep VGA
00:02.0 VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Controller (rev 09)
01:00.0 VGA compatible controller: NVIDIA Corporation GF108GLM [Quadro 1000M] (rev ff)
```

The borkage only happens for me on intel

**Comment by Manishearth** [comment]
`sudo add-apt-repository ppa:oibaf/graphics-drivers` and a dist-upgrade fixed this.

**Comment by kilobtye** [comment]
I digged into this issue and it is deeper than I thought...

It's related to SnapToPixels() in shader code. GL3 version of SnapToPixels uses round(), and I looked up the GL doc:

> round returns a value equal to the nearest integer to x. The fraction 0.5 will round in a direction chosen by the implementation, presumably the direction that is fastest. This includes the possibility that round(x) returns the same value as roundEven(x) for all values of x. 

So if the graphics driver implement round as round to even, point 3.5 and 6.5 will snap to 3 and 7, so the length will become 4 instead of 3. Unfortunately here texts have a lot of x.5 position :(

**Comment by Manishearth** [comment]
cc @glennw could we write our shaders without snaptopixels?

**Comment by metajack** [comment]
If we depend on specific rounding behavior, it sounds like we definitely should not be using that function.

**Comment by mbrubeck** [comment]
Marking as a duplicate of #11751. There is a fix in #11801.
